elastic-mapping-updater/src/main/java/com/microfocus/caf/elastic/tools/ElasticMappingUpdater.java

```diff
@@ -0,0 +1,178 @@
+                getIndexResponse = elasticRequestHandler.getIndex(indexName);
+                indexMappings = getIndexResponse.getMappings().get(indexName);
+                indexTypeMappings = indexMappings.getSourceAsMap();
+                LOGGER.info("------Updated mapping for index '{}': {}", indexName, indexTypeMappings);
+            } catch (final UnsupportedMappingChangesException e) {
+                LOGGER.warn("Unsupported mapping changes for index : {}", indexName, e);
+                // Continue with next index to be updated
+                continue;
+            }
+        }
+    }
+
+    private boolean isMappingChangeSafe(final Map<String, Object> templateMapping, final Map<String, Object> indexMapping)
+            throws JsonProcessingException {
+        final Map<String, Object> ftemplateMapping = FlatMapUtil.flatten(templateMapping);
+        final Map<String, Object> findexMapping = FlatMapUtil.flatten(indexMapping);
+        LOGGER.info("Flattened template mapping : {}", ftemplateMapping);
+        LOGGER.info("Flattened index mapping : {}", findexMapping);
+        final MapDifference<String, Object> diff = Maps.difference(ftemplateMapping, findexMapping);
+        Map<String, ValueDifference<Object>> entriesDiffering = diff.entriesDiffering();
+        if(entriesDiffering.isEmpty())
+        {
+            return true;
+        }
+        else
+        {
+            // ElasticSearch would throw IllegalArgumentException if any such change is included in the index mapping updates
+            entriesDiffering.forEach((key, value) -> LOGGER.info("Entries differing : {}:{}", key, value));
+            return false;
+        }
+    }
+
+    private Map<String, Object> getMappingChanges(final Map<String, Object> templateMapping, final Map<String, Object> indexMapping)
+            throws JsonProcessingException, UnsupportedMappingChangesException {
+        final Map<String, Object> mappingsChanges = new HashMap<>();
+        final MapDifference<String, Object> diff = Maps.difference(templateMapping, indexMapping);
+        final Map<String, ValueDifference<Object>> entriesDiffering = diff.entriesDiffering();
+
+        LOGGER.info("Differing entries: {}", objectMapper.writeValueAsString(entriesDiffering));
+
+        if(!isMappingChangeSafe(templateMapping, indexMapping))
+        {
+            throw new UnsupportedMappingChangesException("Unsupported mapping changes");
+        }
+
+        final Map<String, Object> entriesOnlyInTemplate = diff.entriesOnlyOnLeft();
+        LOGGER.info("Entries only in Template: {}", objectMapper.writeValueAsString(entriesOnlyInTemplate));
+        mappingsChanges.putAll(entriesOnlyInTemplate);
+        final Set<String> fields = entriesDiffering.keySet();
+        for (final String field : fields) {
+            mappingsChanges.put(field, ((ValueDifference<?>) entriesDiffering.get(field)).leftValue());
+        }
+
+        return mappingsChanges;
+    }
+
+}
```

Comment:
The formatting scheme for bracketing methods and if statements seems to be inverted in this class?